src/transport/GroupSession.h

```diff
@@ -18,24 +18,52 @@
 
 #include <app/util/basic-types.h>
 #include <lib/core/GroupId.h>
+#include <lib/core/ReferenceCounted.h>
 #include <lib/support/Pool.h>
 #include <transport/Session.h>
 
 namespace chip {
 namespace Transport {
 
-class GroupSession : public Session
+#ifndef NDEBUG
+class GroupSessionDeleter
 {
 public:
-    GroupSession(GroupId group, FabricIndex fabricIndex, NodeId sourceNodeId) : mGroupId(group), mSourceNodeId(sourceNodeId)
+    static void Release(IncomingGroupSession * entry) {}
+    static void Release(OutgoingGroupSession * entry) {}
+};
+#endif
+
+class IncomingGroupSession : public Session
+#ifndef NDEBUG
+    // The group session is ephemeral, its lifespan is controller by who using it. To prevent the object being destroyed while there
```

Comment:
```suggestion
    // The group session is ephemeral, its lifespan is controlled by whoever is using it. To prevent the object being destroyed while there
```